addons/survey/controllers/main.py

```diff
@@ -573,7 +577,7 @@ def _extract_comment_from_answers(self, question, answers):
     def survey_print(self, survey_token, review=False, answer_token=None, **post):
         '''Display an survey in printable view; if <answer_token> is set, it will
         grab the answers of the user_input_id that has <answer_token>.'''
-        access_data = self._get_access_data(survey_token, answer_token, ensure_token=False, check_partner=False)
+        access_data = self._get_access_data(survey_token, answer_token, ensure_token=False, check_partner=False, check_deadline=False)
```

Comment:
Wouldn't it be easier to not change any method definitions and just do this?

```py
if access_data['validity_code'] === "answer_deadline":
    # ignore validity deadline when printing a survey
    access_data['validity_code'] = True 
```

If the check was ignored multiple times I'd say it's worth it to have an additional method param but it does not seem like it is (and we usually avoid early "in case we need it later", especially for old stable).

Reply:
You're right, it is also commonly done in these routes, including just below, and I didn't see it...
It is really straightforward this way.
My apologies.